### PR TITLE
feat(wmg): move dataset metadata builder to the pipeline

### LINKED ISSUE #5828: tech(cellguide): update get pinned ontology url to pull from a specific release tag in single-cell-curation
[Slack context](https://czi-sci.slack.com/archives/C023Q1APASK/p1695743946165449)

`get_pinned_ontology_url` should be updated to pull from a specific release tag in `single-cell-curation` instead of from `main`. Otherwise, updates to the pinned ontology URLs will break existing unit tests.

### PR DESCRIPTION
**Pull request by atarashansky**
## Reason for Change

- This is the first step towards completing #5585 
- This fixes #5828
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
- remove
- modify

## Testing steps

- Tested locally
- Tested in [rdev](https://pr-5825-frontend.rdev.single-cell.czi.technology/gene-expression)

### CODE REVIEW

File: backend/wmg/data/schemas/data_schema_config.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-WMG_DATA_SCHEMA_VERSION = "v1"
+WMG_DATA_SCHEMA_VERSION = "v2"
```
Comment (atarashansky) [inline backend/wmg/data/schemas/data_schema_config.py:1]: bump data schema version

File: backend/wmg/data/snapshot.py [hunk 2]
```diff
@@ -135,10 +140,9 @@ def load_snapshot(
             snapshot_id=snapshot_id,
             read_versioned_snapshot=read_versioned_snapshot,
         )
-
-    if not hasattr(cached_snapshot, "dataset_dict"):
+    # TODO: Once the pipeline generates the V2 snapshot, this can be removed
+    if cached_snapshot.dataset_metadata is None:
```
Comment (atarashansky) [inline backend/wmg/data/snapshot.py:144]: this backwards compatibility code will be removed after V2 snapshot exists

File: backend/wmg/data/snapshot.py [hunk 3]
```diff
@@ -207,6 +211,9 @@ def _load_snapshot(*, snapshot_schema_version: str, snapshot_id: str, read_versi
     dataset_to_gene_ids = _load_dataset_to_gene_ids_data(snapshot_dir_path)
     filter_relationships = _load_filter_graph_data(snapshot_dir_path)
 
+    # TODO: Once the pipeline generates the V2 snapshot, the ternary can be removed
+    dataset_metadata = _load_dataset_metadata(snapshot_dir_path) if snapshot_schema_version != "v1" else None
```
Comment (atarashansky) [inline backend/wmg/data/snapshot.py:215]: this backwards compatibility ternary will be removed after V2 snapshot exists

**Comment by atolopko-czi** [comment]
How are these issues related to the PR? It's immediately clear to me.
* https://github.com/chanzuckerberg/single-cell-data-portal/issues/5585
* https://github.com/chanzuckerberg/single-cell-data-portal/issues/5828

**Review by prathapsridharan — APPROVED** [review]
LGTM

**Comment by atarashansky** [comment]
@atolopko-czi 
This PR implements the suggestion you gave towards the end of the discussion in #5585, though I probably should have made a separate issue for this.

This PR also updates the pinned ontology url to fix failing cellguide pipeline unit tests (pinned ontology urls in single-cell-curation were updated yesterday, which led to unit unit tests failing in `main` and in this PR).

**Comment by atolopko-czi** [comment]
> @atolopko-czi This PR implements the suggestion you gave towards the end of the discussion in #5585, though I probably should have made a separate issue for this.

I see, ty! The issue title was not informative. :)

> This PR also updates the pinned ontology url to fix failing cellguide pipeline unit tests (pinned ontology urls in single-cell-curation were updated yesterday, which led to unit unit tests failing in `main` and in this PR).

Thanks, I now see that one file change.

**Pull request merged**